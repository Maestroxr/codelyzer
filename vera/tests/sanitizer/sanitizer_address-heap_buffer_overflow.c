#include <stdlib.h>
#include <stdio.h>

int main(int argc, char** argv) 
{
	int res = -1;
	/*
		Address Sanitizer
	*/
	
	int* array = new int[ 100 ];
	array[ 0 ] = 0;
	if (array[ argc+ 100 ])
	{
		// BOOM Heap-buffer-overflow	-V-
		res = array[ argc + 100 ];  // BOOM Heap-buffer-overflow
	}
	delete [] array;
	
	/***/
	return res;
}