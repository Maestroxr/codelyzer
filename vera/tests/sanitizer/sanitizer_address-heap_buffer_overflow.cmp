C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_buffer_overflow.c:13: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_buffer_overflow]: Heap buffer overflow, suspect variables: ['array']
	Was accessed by read of size 4 at:
		#0 sanitizer_address-heap_buffer_overflow.c:13 in function 'main'
	Accessed address is located 4 bytes to the right of 400-byte region, allocated at:
		#1 sanitizer_address-heap_buffer_overflow.c:11 in function 'main'
