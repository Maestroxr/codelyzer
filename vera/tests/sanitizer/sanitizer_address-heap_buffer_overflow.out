C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_buffer_overflow.c:7: Pointer's 'array' malloc is missing null pointer check.
C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_buffer_overflow.c:9: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_buffer_overflow]: Heap buffer overflow, suspect variables: ['array']
	Was accessed by read of size 4 at:
		sanitizer_address-heap_buffer_overflow.c:9 in function 'main'

	Accessed address is located 4 bytes to the right of 400-byte region, allocated at:
		sanitizer_address-heap_buffer_overflow.c:7 in function 'main'

